src/librustdoc/html/static/storage.js

```diff
@@ -29,8 +41,24 @@ function getCurrentValue(name) {
 }
 
 function switchTheme(styleElem, mainStyleElem, newTheme) {
-    styleElem.href = mainStyleElem.href.replace("rustdoc.css", newTheme + ".css");
-    updateLocalStorage('theme', newTheme);
+    var newHref = mainStyleElem.href.replace("rustdoc.css", newTheme + ".css");
+    var found = false;
+
+    if (savedHref.length === 0) {
+        onEach(document.getElementsByTagName("link"), function(el) {
+            savedHref.push(el.href);
+        });
+    }
+    onEach(savedHref, function(el) {
+        if (el === newHref) {
+            found = true;
+            return true;
+        }
+    });
+    if (found === true) {
```

Comment:
I kinda want to see a fallback to "main" here, in case there is something amiss in local storage? Right now it just... does nothing, if this function is called with a name that doesn't match any of the available themes. To the user, this is the same as defaulting to "main", but i'd like to see that codified here as well.

Reply:
It is already the case. The last imported style is the main one. So if we do nothing, the main one will remain.

Reply:
Hmm. What i was really after was "overwrite invalid local storage value with 'main'" but i think i'll just let it slide, in case there's a collision with some other local storage key, or with different versions in case we add new themes later on.